# Phalstedii_Phylogenomics
Collection of scripts for a range of genomic data processing and analysis.
Not everything is documented yet, but most scripts have some helpful information if you type `python script.py -h`.

## Contents

* [Converting SNP coordinates](#Converting-SNP-coordinates)

___
## Converting SNP coordinates
The script `VCFCoordConv.py` converts SNP coordinates (i.e. chromosomes and positions in a VCF file) from one coordinate system to another.

#### Example command
`python VCFCoordConv.py --vcf /path/to/file.vcf.gz' --table /path/to/table.csv'`

`python VCFCoordConv.py -h` Will print a full list of command arguments.

#### Command arguments
| Name | Description |
| :--: | ----------- | 
| `vcf` | Path to the vcf file or gzipped vcf file (required) |
| `table`  | Path to the csv table of the syntenic path generated by SynMap (required) |
| `export`  | Whether the dictionary of coordinates should be exported as a csv table (optional) |

#### Notes
The VCF should not contain the INFO field (as it can contains "," and thus impacts the script outputs). To run it, ensure that you are using Python v.3.7, and have installed the following dependencies: [os](https://docs.python.org/3/library/os.html), X, re, [pandas](https://pandas.pydata.org/), [argparse](https://docs.python.org/3/library/argparse.html), and [sys](https://docs.python.org/3/library/sys.html).
